modules/installation-user-infra-machines-advanced-customizing-live-iscsi-manual.adoc

```diff
@@ -0,0 +1,64 @@
++
+[source,text]
+----
+$ coreos-installer iso customize \
+    --pre-install mount-iscsi.sh \ <1>
+    --post-install unmount-iscsi.sh \ <2>
+    --dest-device /dev/disk/by-path/<IP_address>:<port>-iscsi-<target_iqn>-lun-<lun> \ <3>
+    --dest-ignition config.ign \ <4>
+    --dest-karg-append rd.iscsi.initiator=<initiator_iqn> \ <5>
+    --dest-karg-append netroot=<target_iqn> \ <6>
+    -o custom.iso rhcos-<version>-live.x86_64.iso
+----
+<1> The script that gets run before installation. It should contain the `iscsiadm` commands for mounting the iSCSI target and any commands enabling multipathing.
+<2> The script that gets run after installation. It should contain the command `iscsiadm --mode node --logout=all`.
+<3> The location of the destination system. You must provide the IP address of the target portal, the associated port number, the target iSCSI node in IQN format, and the iSCSI logical unit number (LUN).
+<4> The ignition configuration for the destination system.
+<5> The iSCSI initiator, or client, name in IQN format. The initiator forms a session to connect to the iSCSI target.
+<6> The the iSCSI target, or server, name in IQN format.
+endif::[]
+
+ifeval::["{boot-media}" == "PXE environment"]
+. Retrieve the {op-system} `kernel`, `initramfs` and `rootfs` files from the link:https://mirror.openshift.com/pub/openshift-v4/dependencies/rhcos/latest/[{op-system} image mirror] page and run the following command to create a new customized `initramfs` file with the following information:
++
+[source,text]
+----
+$ coreos-installer pxe customize \
+    --pre-install mount-iscsi.sh \ <1>
+    --post-install unmount-iscsi.sh \ <2>
+    --dest-device /dev/disk/by-path/<IP_address>:<port>-iscsi-<target_iqn>-lun-<lun> \ <3>
+    --dest-ignition config.ign \ <4>
+    --dest-karg-append rd.iscsi.initiator=<initiator_iqn> \ <5>
+    --dest-karg-append netroot=<target_iqn> \ <6>
+    -o custom.img rhcos-<version>-live-initramfs.x86_64.img
+----
+<1> The script that gets run before installation. It should contain the `iscsiadm` commands for mounting the iSCSI target and any commands enabling multipathing.
+<2> The script that gets run after installation. It should contain the command `iscsiadm --mode node --logout=all`.
+<3> The location of the destination system. You must provide the IP address of the target portal, the associated port number, the target iSCSI node in IQN format, and the iSCSI logical unit number (LUN).
+<4> The ignition configuration for the destination system.
+<5> The iSCSI initiator, or client, name in IQN format. The initiator forms a session to connect to the iSCSI target.
+<6> The the iSCSI target, or server, name in IQN format.
+endif::[]
++
+For more information about the iSCSI options supported by `dracut`, see the link:https://www.man7.org/linux/man-pages/man7/dracut.cmdline.7.html[`dracut.cmdline` manual page].
```

Comment:
same, I love the construction: it's clear and easy to read in both source and the end result.

Reply:
thanks James! 😄